packages/frontend-core/src/components/grid/stores/rows.js

```diff
@@ -519,4 +535,15 @@ export const initialise = context => {
       })
     }
   })
+
+  // Ensure any unsaved changes are saved when changing cell
+  previousFocusedCellId.subscribe(id => {
+    const rowId = id?.split("-")[0]
+    const hasErrors = validation.actions.rowHasErrors(rowId)
+    const hasChanges = Object.keys(get(rowChangeCache)[rowId] || {}).length > 0
+    const isSavingChanges = get(inProgressChanges)[rowId]
+    if (rowId && !hasErrors && hasChanges && !isSavingChanges) {
```

Comment:
The `rowChangeCache` stores changes to rows that have yet to be saved to the server - this typically is empty unless your row has an error, or now also on the first key press of a newly focused cell.

`inProgressChanges` is a store to keep track of which row IDs have in flight requests. As in an API request has been made to the server, but we haven't got a response yet. Since latency is variable we can't always assume this is fast, so we've always kept track of this to ensure we don't double up on requests and cause conflicts.